app/core/rate_limiting/gemini_limiter.py

```
@@ -0,0 +1,335 @@
+                pro_circuit=pro_circuit,
+                total_requests_today=total_requests_today,
+                total_requests_this_minute=total_requests_this_minute,
+                uptime_percentage=uptime_percentage,
+                last_updated=now
+            )
+            
+        except Exception as e:
+            self.logger.error(f"Failed to get usage stats: {e}")
+            raise GeminiServiceUnavailableException(
+                f"Unable to retrieve usage statistics: {e}"
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)